### PR TITLE
[#56011] reset user tokens when refreshing token fails

### PR DESCRIPTION
**Pull request by Kharonus**
[#56011](https://community.openproject.org/work_packages/56011)

### WAT?

- handle timeouts in token refreshes, too
- handle nextcloud brute force throttling

### CODE REVIEW

File: modules/storages/app/common/storages/peripherals/storage_interaction/authentication_strategies/oauth_user_token.rb [hunk 1]
```diff
@@ -124,6 +124,35 @@ def update_refreshed_token(token, http_session)
             token.update(access_token:, refresh_token:)
           end
 
+          def log_and_destroy_token(token, log_message)
```
Comment (judithroth) [inline modules/storages/app/common/storages/peripherals/storage_interaction/authentication_strategies/oauth_user_token.rb]: This method seems unused - you probably planned on using it in `handle_http_error_on_refresh` and `handle_timeout_on_refresh`?

**Comment by judithroth** [comment]
I guess it would be good if someone else also reviews this :)

File: modules/storages/app/common/storages/peripherals/storage_interaction/authentication_strategies/oauth_user_token.rb (hunk 1 above)
Comment (Kharonus) [reply modules/storages/app/common/storages/peripherals/storage_interaction/authentication_strategies/oauth_user_token.rb]: indeed, will delete it.

**Review by mereghost — APPROVED** [review]
LGTM

**Review by judithroth — APPROVED** [review]

**Pull request merged**